src/builder/commands/copy.rs

```diff
@@ -104,6 +111,15 @@ impl BuildStep for Copy {
             // inside the container which is ugly
         }
         hash.field("path", self.path.to_str().unwrap());
+        if let Some(uid) = self.owner_uid {
+            hash.field("owner_uid", uid.to_string());
+        }
+        if let Some(gid) = self.owner_gid {
+            hash.field("owner_gid", gid.to_string());
+        }
+        if let Some(umask) = self.umask {
+            hash.field("umask", umask.to_string());
```

Comment:
Is there any need of hashing configuration parameters instead of hashing actual mode of the target file?